htdocs/api/v0.0.3-dev/candidates/visits/Dicoms.php

```diff
@@ -0,0 +1,148 @@
+ *
+ * @category Main
+ * @package  API
+ * @author   Mouna Safi-Harab <mouna.safiharab@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace Loris\API\Candidates\Candidate\Visit;
+require_once '../Visit.php';
+
+/**
+ * Handles API requests for the candidate's dicom files. Extends
+ * Visit so that the constructor will validate the candidate
+ * and visit_label portion of URL automatically.
+ *
+ * @category Main
+ * @package  API
+ * @author   Mouna Safi-Harab <mouna.safiharab@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Dicoms extends \Loris\API\Candidates\Candidate\Visit
+{
+    /**
+     * Construct a Dicoms class object to serialize candidate visit dicoms
+     *
+     * @param string $method     The method of the HTTP request
+     * @param string $CandID     The CandID to be serialized
+     * @param string $VisitLabel The visit label to be serialized
+     */
+    public function __construct($method, $CandID, $VisitLabel)
+    {
+        $requestDelegationCascade = $this->AutoHandleRequestDelegation;
+
+        $this->AutoHandleRequestDelegation = false;
+
+        if (empty($this->AllowedMethods)) {
+            $this->AllowedMethods = ['GET'];
+        }
+        $this->CandID     = $CandID;
+        $this->VisitLabel = $VisitLabel;
+
+        parent::__construct($method, $CandID, $VisitLabel);
+
+        if ($requestDelegationCascade) {
+            $this->handleRequest();
+        }
+
+    }
+
+    /**
+     * Handles a GET request
+     *
+     * @return none, but populates $this->JSON
+     */
+    public function handleGET()
+    {
+        $this->JSON = [
+                       'Meta' => [
+                                  'CandID' => $this->CandID,
+                                  'Visit'  => $this->VisitLabel,
+                                 ],
+                      ];
+        $this->JSON['DicomTars'] = $this->getVisitDicoms();
+
+    }
+
+    /**
+     * Gets a list of Tars for this visit. Tarname only.
+     *
+     * @return array of strings of Tarnames
+     */
+    function getVisitDicoms()
+    {
+        $factory   = \NDB_Factory::singleton();
+        $DB        = $factory->database();
+        $cand_info = $DB->pselectRow(
+            "SELECT c.PSCID, c.Entity_type
+                FROM candidate c
+                    JOIN session s ON (c.CandID=s.CandID)
+                WHERE c.Active='Y' AND s.Active='Y'
+                AND c.CandID=:CID
+                AND s.Visit_label=:VL",
+            [
+             'CID' => $this->CandID,
+             'VL'  => $this->VisitLabel,
+            ]
+        );
+        if ($cand_info['Entity_type'] == 'Scanner') {
+            $ID            = ":PVL";
+            $params['PVL'] = $this->VisitLabel;
+        } else {
+            $ID = "LOWER(CONCAT(:PPSCID, '_', :PCandID, '_', :PVL, '%'))";
```

Comment:
nitpicky but should there be an _ between `:PVL` and `%` since we only allow PSCID_CANDID_VISITLABEL_* and not PSCID_CANDID_VISITLABEL*

Reply:
also another not so big deal but i am curious as to why you're using `LOWER(...)` here, since value matching in columns tends to be case insensitive and most PSCIDs and VL's (at least one's i've seen) are all uppercase
sorry if this is a stupid question 😝

Reply:
I believe our MRI code some time ago was not case sensitive for the visits (IBIS has visits that are either V3 or v3); I know the DICOM archive module does it this way, perhaps for these cases? not sure